sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java

```diff
@@ -364,6 +381,10 @@ public Read<T> withAutoScaler(AutoScaler autoScaler) {
       return builder().setAutoScaler(autoScaler).build();
     }
 
+    public Read<T> withCloseTimeout(long closeTimeout) {
```

Comment:
Do you want to go with a wait on how long we keep sessions around?
```suggestion
    /** The amount of time to wait before a message that was read has its acknowledgement expire and be returned back to the JMS broker. */
    public Read<T> withMessageAckExpiry(Duration ackExpiry) {
```

If not then I would suggest updating this to:
```suggestion
    /** The amount of time to wait for bundle finalization callbacks allowing for messages to be acknowledged after the reader is closed. */
    public Read<T> withConnectionCloseTimeout(Duration closeTimeout) {
```

Reply:
Yes we want to want to wait before closing the session, and then the connection. The closing of the session will indicate to the broker that all unacknowledged messages must be kept into the queue to be redelivered. 
Don't know which is best, withSessionConnectionCloseTimeou, withSessionCloseTimeout....